gemini-app/src/Components/GCP/ImageViewer.js

```diff
@@ -131,18 +135,20 @@ function ImageViewer({ open, onClose, item, activeTab, platform, sensor }) {
     // update state variables
     useEffect(() => {
         if (open) {
+            // reset image viewer
             setImageViewerReady(false);
-            console.log("prepGcpFilePath initial: ", prepGcpFilePath);
+
+            // get gcp path info
             setSelectedDateGCP(item.date);
             setSelectedPlatformGCP(platform);
             setSelectedSensorGCP(sensor);
-            //fetchAndSetGcpFilePath();
-            if (prepGcpFilePath != "") {
-                setImageViewerReady(true);
-                setIsImageViewerOpen(true);
-            } else {
-                setDialogOpen(true);
-            }
+
+            // check if gcp file exists
+            fetchAndSetGcpFilePath();
+            console.log("prepGcpFilePath initial: ", prepGcpFilePath);
+
+            // open dialog
+            setDialogOpen(true);
```

Comment:
Always display dialog box even when gcp file has been uploaded. However, the button will change if gcp file does/does not exist.

```
if (gcpLocationsFile) {
            const newPath = `Raw/${selectedYearGCP}/${selectedExperimentGCP}/${selectedLocationGCP}/${selectedPopulationGCP}/${gcpLocationsFile}`;
            setPrepGcpFilePath(newPath);
            setButtonLabel("Continue with current GCP");
            console.log("GCP path found, setting to ", newPath);
        } else {
            setPrepGcpFilePath("");
            setButtonLabel("Continue without GCP");
            console.log("No GCP path found");
        }
```

Reply:
<img width="487" alt="Screenshot 2024-12-17 at 3 04 06 PM" src="https://github.com/user-attachments/assets/090870f7-2ea3-490d-ab2b-a0c25bf114cb" />